### DISCUSSION TITLE
Examples for using transform

### DISCUSSION THREAD

**Discussion by meenar-se** [sensedeep/dynamodb-onetable#50] [Q&A] [answered]
Can you please share any example of using the schema attribute property  **transform**?

I have a use case where I need to form composite key value for gsi(mail:flagged) based on the Boolean attribute isFlagged.

**Comment by mobsense** [comment, marked as answer]
You would use a value template string or function rather than the transform property.

The transform property is designed for assisting with database migrations where you need to do one-off custom processing when reading and writing.

A value template can be like:

```
            gs1pk:          { type: String, value: 'account-email:${email}' },
            gs1sk:          { type: String, value: 'account:' },
            email:          { type: String, required: true },
```

Here the gs1pk is defined based on the property "email".

You can also set the value to a function:

```
            gs1pk:          { type: String, value: (name, context, properties) => {
                                  // Do anything here you want.
                                  if (properties.isFlagged) {
                                      return 'something'
                                  } else {
                                      return properties.isFlagged.toLocaleString()
                                  }
                            }
```

**Reply by meenar-se** [reply]
It’s working. Thank you for timely response
